### ISSUE TITLE
Adding routes to serve.py

### ISSUE THREAD

**Issue by areeves87** [uber/ludwig#480]
**Is your feature request related to a problem? Please describe.**
I'd like to check whether the server is up or down (ie a healthcheck) with a simple GET request to the root path `/` of the ludwig server.

In general it would be nice to have some instructions/examples for how to extend the functionality of serve.py.

**Describe the use case**
This new route would help run health checks on a remote instance of the ludwig server. In my case it would be for an aws ecs cluster, which requires a valid URI with no spaces for healthchecks. It is not straight forward to satisfy this criteria with the `/predict` route.

**Describe the solution you'd like**
Add a simple `GET` route in serve.py's fastApi for the root path `/`, but have not had any success yet.

**Describe alternatives you've considered**

**Additional context**
Here's what I've edited in `ludwig/serve.py` before building the docker image. 

```python
def server(model):
    app = FastAPI()

    input_features = {
        f['name'] for f in model.model_definition['input_features']
    }

    @app.get('/')
    def root():
        return JSONResponse({"message": "ludwig is up!"})

    @app.post('/predict')
    async def predict(request: Request):
        form = await request.form()
        files, entry = convert_input(form)

        try:
            if (entry.keys() & input_features) != input_features:
                return JSONResponse(ALL_FEATURES_PRESENT_ERROR,
                                    status_code=400)
            try:
                resp = model.predict(data_dict=[entry]).to_dict('records')[0]
                return JSONResponse(resp)
            except Exception as e:
                logger.error("Error: {}".format(str(e)))
                return JSONResponse(COULD_NOT_RUN_INFERENCE_ERROR,
                                    status_code=500)
        finally:
            for f in files:
                os.remove(f.name)


    return app
```

And what I changed in the Dockerfile to make sure it uses the updated serve.py 
```
FROM tensorflow/tensorflow:latest-py3

RUN apt-get install -y --no-install-recommends \
        git libgmp3-dev
RUN git clone --depth=1 https://github.com/uber/ludwig.git \
    && cd ludwig/ \
    && pip install -r requirements.txt \
    && python -m spacy download en \
    && python setup.py install

COPY ludwig/serve.py ludwig/build/lib/ludwig/serve.py
COPY ludwig/serve.py ludwig/ludwig/serve.py

WORKDIR /data

ENTRYPOINT ["ludwig"]
```

**Comment by msaisumanth** [comment]
@johnwahba FYI

**Comment by wahba-stripe** [comment]
Change seems super reasonable. If you build it and add a test for the healthcheck I'd have no objections.

**Comment by w4nderlust** [comment]
@areeves87 thanks, this makes sense!
The only thing that's not clear to me is:
```
COPY ludwig/serve.py ludwig/build/lib/ludwig/serve.py
COPY ludwig/serve.py ludwig/ludwig/serve.py
```
Why do you need them?

**Comment by areeves87** [comment]
Okay great - I made the PR!

@w4nderlust those two lines are back from when I didn't understand how to add a new route to server.py. In hindsight, I agree they do not make sense. 

Eventually, I figured out how to add new routes. But some better documentation on how to extend ludwig's functionality would be helpful. What I eventually realized was that one has to call `python setup.py install` to implement one's changes to ludwig.
